objectbox/lib/flatbuffers/flat_buffers.dart

```diff
@@ -703,18 +703,47 @@ class Builder {
   }
 
   int _writeString(String value) {
-    // TODO(scheglov) optimize for ASCII strings
-    List<int> bytes = utf8.encode(value);
-    int length = bytes.length;
+    // [utf8.encode()] is slow (up to at least Dart SDK 2.13). If the given
```

Comment:
Dart sources are available in the [SDK under BSD license](https://github.com/dart-lang/sdk/blob/master/sdk/lib/convert/utf.dart#L88) and could be adapted. Still not going to be faster than plain ascii writing because it needs the at least one additional allocation for the maximum utf-8 long buffer, right?